src/lib/eliom_form.eliom

```diff
@@ -0,0 +1,337 @@
+(* Ocsigen
+ * http://www.ocsigen.org
+ * Module Eliom_predefmod
+ * Copyright (C) 2007 Vincent Balat
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, with linking exception;
+ * either version 2.1 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+*)
+
+{shared{
+
+module type Html5_core = sig
+
+  include Html5_sigs.T
+    with type 'a Xml.W.t = 'a
+     and type 'a Xml.W.tlist = 'a list
+
+  type ('a, 'b, 'c) lazy_star =
+    ?a: (('a attrib) list) ->
+    ('b elt) list Eliom_lazy.request ->
+    'c elt
+
+  val lazy_form:
+    ([< Html5_types.form_attrib ],
+     [< Html5_types.form_content_fun ],
+     [> Html5_types.form ]) lazy_star
+
+  val uri_of_fun : (unit -> string) -> Xml.uri
```

Comment:
I know you need it for compat but, hum, why aren't we using `Xml.uri_of_string` here ?